borgmatic/commands/borgmatic.py

```diff
@@ -104,6 +107,9 @@ def main():  # pragma: no cover
                     )
                 if args.check:
                     check.check_archives(args.verbosity, repository, consistency, remote_path=remote_path)
+
+            hook.exec_cmd(hooks.get('after_backup'))
     except (ValueError, OSError, CalledProcessError) as error:
+        hook.exec_cmd(hooks.get('on_error'))
```

Comment:
Very good catch! I actually overlooked this...

I'd to propose to add a separate ```try / except``` clause enclosing everythin between the ```before_backup``` and ```after_backup``` hook. This would enable a distinct ```on_error``` hook per config file.

IMHO, accumulating all ```on_error``` hooks would contradict the fact that they can be defined individually for each config file.  Dropping ```on_error``` would probably prevent some use cases, such as getting a targeted notification in case a backup process has failed.

Reply:
The `try`/`except` option as you describe sounds good. However, if you don't `raise` there, it would be good to avoid swallowing error information. For instance, without a `raise`, borgmatic could encounter an error, you'd get notified of the error (via an `on_error` script), but then you'd have no further information about what went wrong nor any way to debug the problem.

One way I could see dealing with this is adding some logging of the error information, so the information isn't lost. But that might be blocked on https://tree.taiga.io/project/witten-borgmatic/issue/33 (which I've actually started working on).

Another option would be to `raise` for now, and just say that for the time being, an error in one config file causes borgmatic to immediately exit, which incidentally is the current behavior. We can always revise/improve that in a subsequent release.